docs/reference/esql/functions/kibana/definition/bucket.json

```diff
@@ -34,246 +34,6 @@
       "variadic" : false,
       "returnType" : "datetime"
     },
-    {
-      "params" : [
```

Comment:
@drewdaemon I've just realized you'll need this too. Here's the deal - we have automatic promotion of string literals. For this particular function these always had to be literals anyway so we no longer think of these as valid signatures.

This is an interesting wrinkle. We don't send anything telling you this has to be a literal. We probably should.

Reply:
@nik9000 thanks for thinking of us! Yes, we currently have this information [hard-coded on our end](https://github.com/elastic/kibana/blob/f018af0d96e25fdc0c257534159e77b0a1975039/packages/kbn-esql-validation-autocomplete/src/definitions/functions.ts#L572-L573). If there's a way to surface this in the definitions y'all provide us... all the better!

Reply:
We could generate something like:
```
       {
          "name" : "from",
          "type" : "keyword",
          "optional" : false,
          "description" : "",
          "only_literals": "date"
        },
```

Or something like that. But in general if something says `datetime` now it can take a literal string that is formatted like a date. Same for numbers. But that's quite a lot to add to your editor, especially right now.

There are other cases where we send you `"only_literals": "any"` for bits that take any literal. The `buckets` parameter here only takes literal `integer`s. No fields allowed.

Reply:
Yeah, this kind of thing would be great. I have put https://github.com/elastic/kibana/issues/179634 a bit on hold while I squish editor bugs for the 8.14 release. Once that settles down, I'm going to work on pulling these generated definitions into Kibana "for real."

At that point, I think I'll have a clearer understanding of exactly what structure would work best.

> But in general if something says datetime now it can take a literal string that is formatted like a date. Same for numbers. 

Noted, thanks!